File: generator/src/test/resources/swift/expected/RequestMethodsTest/test-request-method-generation-with-nullify.output.swift
import Sunday

public final class API<TransportType : Transport> : Sendable {

  public static var problemTypes: [ProblemRegistration] {
    return [
      ProblemRegistration(type: TestNotFoundProblem.type, problemType: TestNotFoundProblem.self),
      ProblemRegistration(type: AnotherNotFoundProblem.type, problemType: AnotherNotFoundProblem.self)
    ]
  }
  public let transport: TransportType
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    transport: TransportType,
    defaultContentTypes: [MediaType] = [],
    defaultAcceptTypes: [MediaType] = [.json],
    problemTypes: [ProblemRegistration] = API.problemTypes
  ) {
    self.transport = transport
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
    problemTypes.forEach { $0.register(on: transport) }
  }

  public func fetchTest1(limit: Int) throws -> Sunday.NilableOperation<Empty, Test, TransportType> {
    return Sunday.NilableOperation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .get,
        pathTemplate: "/test1",
        pathParameters: nil,
        queryParameters: [
          "limit": try ParameterValues.encode(limit)
        ],
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      ),
      nilify: Sunday.NilifySpec(
        statuses: [404, 405],
        problemTypes: [TestNotFoundProblem.self, AnotherNotFoundProblem.self]
      )
    )
  }

  public func fetchTest2(limit: Int) throws -> Sunday.NilableOperation<Empty, Test, TransportType> {
    return Sunday.NilableOperation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .get,
        pathTemplate: "/test2",
        pathParameters: nil,
        queryParameters: [
          "limit": try ParameterValues.encode(limit)
        ],
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      ),
      nilify: Sunday.NilifySpec(
        statuses: [404],
        problemTypes: [TestNotFoundProblem.self, AnotherNotFoundProblem.self]
      )
    )
  }

  public func fetchTest3(limit: Int) throws -> Sunday.NilableOperation<Empty, Test, TransportType> {
    return Sunday.NilableOperation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .get,
        pathTemplate: "/test3",
        pathParameters: nil,
        queryParameters: [
          "limit": try ParameterValues.encode(limit)
        ],
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      ),
      nilify: Sunday.NilifySpec(
        statuses: [],
        problemTypes: [TestNotFoundProblem.self, AnotherNotFoundProblem.self]
      )
    )
  }

  public func fetchTest4(limit: Int) throws -> Sunday.NilableOperation<Empty, Test, TransportType> {
    return Sunday.NilableOperation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .get,
        pathTemplate: "/test4",
        pathParameters: nil,
        queryParameters: [
          "limit": try ParameterValues.encode(limit)
        ],
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      ),
      nilify: Sunday.NilifySpec(
        statuses: [404, 405],
        problemTypes: []
      )
    )
  }

  public func fetchTest5(limit: Int) throws -> Sunday.NilableOperation<Empty, Test, TransportType> {
    return Sunday.NilableOperation(
      transport: self.transport,
      spec: Sunday.OperationSpec(
        method: .get,
        pathTemplate: "/test5",
        pathParameters: nil,
        queryParameters: [
          "limit": try ParameterValues.encode(limit)
        ],
        body: Empty.none,
        contentTypes: nil,
        acceptTypes: self.defaultAcceptTypes,
        headers: nil
      ),
      nilify: Sunday.NilifySpec(
        statuses: [404],
        problemTypes: []
      )
    )
  }

}

File: generator/src/test/resources/kotlin/expected/sunday/RequestMethodsTest/test-request-method-generation-with-nullify.output.kt
package io.test.service

import io.outfoxx.sunday.MediaType
import io.outfoxx.sunday.NullableOperation
import io.outfoxx.sunday.NullifySpec
import io.outfoxx.sunday.OperationSpec
import io.outfoxx.sunday.Transport
import io.outfoxx.sunday.http.Method
import io.outfoxx.sunday.http.Request
import io.outfoxx.sunday.nullableOperation
import io.test.AnotherNotFoundProblem
import io.test.Test
import io.test.TestNotFoundProblem
import kotlin.Int
import kotlin.Unit
import kotlin.collections.List

public class API<Req : Request>(
  public val transport: Transport<Req>,
  public val defaultContentTypes: List<MediaType> = listOf(),
  public val defaultAcceptTypes: List<MediaType> = listOf(MediaType.JSON),
) {
  init {
    transport.registerProblem("http://example.com/test_not_found", TestNotFoundProblem::class)
    transport.registerProblem("http://example.com/another_not_found", AnotherNotFoundProblem::class)
  }
  public fun fetchTest1(limit: Int): NullableOperation<Unit, Test, Req> =
      this.transport.nullableOperation<Unit, Test, Req>(
    OperationSpec(
      method = Method.Get,
      pathTemplate = "/test1",
      queryParameters = mapOf(
        "limit" to limit
      ),
      acceptTypes = this.defaultAcceptTypes
    ),
    NullifySpec(
      statuses = listOf(404, 405),
      problemTypes = listOf(TestNotFoundProblem::class, AnotherNotFoundProblem::class)
    )
  )

  public fun fetchTest2(limit: Int): NullableOperation<Unit, Test, Req> =
      this.transport.nullableOperation<Unit, Test, Req>(
    OperationSpec(
      method = Method.Get,
      pathTemplate = "/test2",
      queryParameters = mapOf(
        "limit" to limit
      ),
      acceptTypes = this.defaultAcceptTypes
    ),
    NullifySpec(
      statuses = listOf(404),
      problemTypes = listOf(TestNotFoundProblem::class, AnotherNotFoundProblem::class)
    )
  )

  public fun fetchTest3(limit: Int): NullableOperation<Unit, Test, Req> =
      this.transport.nullableOperation<Unit, Test, Req>(
    OperationSpec(
      method = Method.Get,
      pathTemplate = "/test3",
      queryParameters = mapOf(
        "limit" to limit
      ),
      acceptTypes = this.defaultAcceptTypes
    ),
    NullifySpec(
      statuses = listOf(),
      problemTypes = listOf(TestNotFoundProblem::class, AnotherNotFoundProblem::class)
    )
  )

  public fun fetchTest4(limit: Int): NullableOperation<Unit, Test, Req> =
      this.transport.nullableOperation<Unit, Test, Req>(
    OperationSpec(
      method = Method.Get,
      pathTemplate = "/test4",
      queryParameters = mapOf(
        "limit" to limit
      ),
      acceptTypes = this.defaultAcceptTypes
    ),
    NullifySpec(
      statuses = listOf(404, 405),
      problemTypes = listOf()
    )
  )

  public fun fetchTest5(limit: Int): NullableOperation<Unit, Test, Req> =
      this.transport.nullableOperation<Unit, Test, Req>(
    OperationSpec(
      method = Method.Get,
      pathTemplate = "/test5",
      queryParameters = mapOf(
        "limit" to limit
      ),
      acceptTypes = this.defaultAcceptTypes
    ),
    NullifySpec(
      statuses = listOf(404),
      problemTypes = listOf()
    )
  )
}
